logstash-core/lib/logstash/logging/logger.rb

```diff
@@ -6,6 +6,7 @@ class Logger
       java_import org.apache.logging.log4j.Level
       java_import org.apache.logging.log4j.LogManager
       java_import org.apache.logging.log4j.core.config.Configurator
+      @@config_mutex = Mutex.new
```

Comment:
The problem is log4j changed to the running configuration are not thread safe.

@talevy pointed me to the official log4j doc for this, see https://logging.apache.org/log4j/2.x/manual/customconfig.html


```
Applications sometimes have the need to customize logging separate from the actual configuration. Log4j allows this although it suffers from a few limitations:

If the configuration file is changed the configuration will be reloaded and the manual changes will be lost.
Modification to the running configuration requires that all the methods being called (addAppender and addLogger) be synchronized.
``